usefulUtilities/bouncerZone/bouncerZone.js

```diff
@@ -55,140 +99,84 @@
         if (DEBUG) {
             print("You MAY NOT enter - verification failed for entity: " + entityID);
         }
-        WalletScriptingInterface.ownershipVerificationSuccess.disconnect(ticketVerificationSuccess);
-        WalletScriptingInterface.ownershipVerificationFailed.disconnect(ticketVerificationFailed);
-
+        if (signalsConnected) {
+            WalletScriptingInterface.ownershipVerificationSuccess.disconnect(ticketVerificationSuccess);
+            WalletScriptingInterface.ownershipVerificationFailed.disconnect(ticketVerificationFailed);
+        }
         bounceAvatarFromZone();
     }
 
     // Searches around MyAvatar for a wearable whose marketplaceID matches ticketMarketplaceID.
     var WEARABLE_SEARCH_RADIUS = 10;
     var potentialTicketEntityID;
-    function searchForTicketWearable() {
+    var signalsConnected = false;
+    function userHasTicketWearable() {
         potentialTicketEntityID = "";
         var currentAvatarWearableIDs = Entities.findEntitiesByType('Model', MyAvatar.position, WEARABLE_SEARCH_RADIUS);
 
         for (var i = 0; i < currentAvatarWearableIDs.length; i++) {
             var properties = Entities.getEntityProperties(currentAvatarWearableIDs[i],
                 ['marketplaceID', 'certificateID', 'parentID']);
             if (properties.marketplaceID === ticketMarketplaceID && properties.parentID === MyAvatar.sessionUUID) {
-                WalletScriptingInterface.ownershipVerificationSuccess.connect(ticketVerificationSuccess);
-                WalletScriptingInterface.ownershipVerificationFailed.connect(ticketVerificationFailed);
+                if (!signalsConnected) {
+                    WalletScriptingInterface.ownershipVerificationSuccess.connect(ticketVerificationSuccess);
+                    WalletScriptingInterface.ownershipVerificationFailed.connect(ticketVerificationFailed);
+                    signalsConnected = true;
+                }
                 potentialTicketEntityID = currentAvatarWearableIDs[i];
                 WalletScriptingInterface.proveAvatarEntityOwnershipVerification(currentAvatarWearableIDs[i]);
-                return;
+                return true;
             }
         }
-
-        bounceAvatarFromZone();
+        
+        return false;
     }
 
     // Updates some internal variables based on the current userData property of the attached zone entity.
     function updateParametersFromUserData() {
-        var userDataProperty = {
-            "whitelist" : {
-                "rejectTeleportLocation" : "/",
-                "marketplaceID" : "",
-                "usernames" : [""]
-            }
-        };
+        var userDataProperty = DEFAULT_USER_DATA;
```

Comment:
since you have this, why do you need the `|| XYZ` clauses below?

Reply:
the userDataProperty gets overwritten by the userData read from the entity, so this ensures that each value exists even if any were left blank in the userData. I think we could add more error checking in the future such as validating the rejectionLocation or soundURL but, for now, we at least make sure we are not trying to send the user to a null position.

Reply:
oh, good point! in that case, could you please replace your `|| XYZ` clauses with `DEFAULT_USER_DATA.XYZ` statements?

Reply:
yep